examples/custom_404.rs

```diff
@@ -0,0 +1,38 @@
+extern crate iron;
+extern crate router;
+
+// To build, $ cargo test
+// To use, go to http://127.0.0.1:3000/test
+
+use iron::{Iron, Request, Response, IronResult, AfterMiddleware, Chain};
+use iron::error::{IronError};
+use iron::status;
+use router::{Router};
+
+struct Custom404;
+
+impl AfterMiddleware for Custom404 {
+
+    fn catch(&self, _: &mut Request, err: IronError) -> IronResult<Response> {
+        println!("Hitting custom 404 middleware");
+
+        match err.response.status {
```

Comment:
Ah interesting, this is one way to do it.

Originally what I had in mind was catching the specific `NoRoute` error thrown by router using `err.error.is::<NoRoute>()`, but this works too. Using `is` has the nice benefit of demonstrating how to disambiguate the error in an `IronError`, but if the goal is just to send a custom 404 then this is clearer. What do you think?